docs/source/architecture/slashing.rst

```diff
@@ -0,0 +1,150 @@
+The Slashing Protocol
+=====================
+
+TBD
+
+
+Violations
+----------
+
+TBD
+
+
+Calculating the slashing penalty
+--------------------------------
+
+TBD (https://github.com/nucypher/nucypher/issues/803)
+
+
+How slashing affects stake
+--------------------------
+
+The goal of slashing is to reduce the number of tokens that belongs to a staking offender.
+In this case, the main task is not to violate the logic of locking tokens.
+The entire stake consists of:
+
+    * tokens which the staker can withdraw at any moment
+    * tokens locked for a specific period
+
+A staker may extend the unlock period for any number of portions of their total stake. This divides the stake into smaller parts, each with a unique unlock date in the future. Stakers may also acquire and lock new tokens. The total stake is represented as the sum of all the different sub-stakes active in a given cycle (new cycle every 24h), which includes locked sub-stakes, and any sub-stakes that have passed their unlock date, and can be freely withdrawn. Each sub-stake has a beginning and duration (lock time). When a staker confirms activity each day, the remaining lock time for relevant sub-stakes is reduced. 
+
+Sub stakes get slashed in the order of their remaining lock time, beginning with the shortest – so the first portion of the stake to be slashed is the unlocked portion. After that, if necessary, locked sub-stakes are decreased – the shortest sub stake is decreased by the required amount; if the adjustment of that sub-stake is insufficient to fulfil the required punishment sum, then the next shortest sub-stake is decreased, and so on. Sub-stakes that begin in the next period are checked separately.
```

Comment:
I like the language of this and previous paragraph. Very clear and understandable!

Reply:
A bit of a run-on sentence here: 

"The total stake is represented as the sum of all the different sub-stakes active in a given cycle (new cycle every 24h), which includes locked sub-stakes, and any sub-stakes that have passed their unlock date, and can be freely withdrawn."

How about:

"The total stake value is the sum of all active sub-stake values in a given period (new period every 24h).  This includes locked sub-stakes and any sub-stakes that have passed their unlock date which can be freely withdrawn."

Reply:
I realized that meaning of "stake" is a little bit different in parts of this doc. Is it all tokens that staker put as a stake in the contract and doesn't matter them are locked or not in the current period? Or only locked tokens are "stake"?

Reply:
Hmm - I think.. only locked tokens. Perhaps we need a term for the tokens in escrow that are not locked?

Reply:
@KPrasch In that case, it seems a little odd that unlocked tokens sitting around can be slashed for misdeeds, but do not count at all towards your total stake (and commensurate payouts). If you stop earning fees/rewards on tokens once they reach their unlock date, then there is zero upside to leaving them in that state. If we want to rush stakers into re-staking the tokens for a new lock period, or withdraw asap – which makes sense from a user point of view, as stakes with no lock period can unpredictably disappear, which may impede sharing policies – then I suppose this is the right approach. No matter what, these rules need to be extremely clear to stakers.